src/components/ExpandableDataTable/ExpandableDataTable.jsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import {
+  DataTable,
+  Table,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableExpandedRow,
+  TableExpandHeader,
+  TableExpandRow,
+} from "carbon-components-react";
+import PropTypes from "prop-types";
+import "./ExpandableDataTable.scss";
+
+const ExpandableDataTable = (props) => {
+  const { rows, headers, additionalData, component, useZebraStyles } = props;
+
+  const renderExpandableRowComponent = (row, additionalData) => {
+    const filteredData = additionalData.filter((data) => row.id === data.id);
+    return filteredData.map((data) => component(data));
+  };
+
+  return (
+    <DataTable
+      rows={rows}
+      headers={headers}
+      additionalData={additionalData}
+      useZebraStyles={useZebraStyles}
+      render={({
+        rows,
+        headers,
+        getHeaderProps,
+        getRowProps,
+        getTableProps,
+        getExpandHeaderProps,
+      }) => (
+        <>
```

Comment:
there is no need of having this fragment as <Table> itself wraps all the underlying components